http_post.go

```diff
@@ -68,18 +95,40 @@ func (h *HTTPPostNode) postRow(group models.GroupID, row *models.Row) {
 	result := new(models.Result)
 	result.Series = []*models.Row{row}
 
-	body := h.bp.Get()
-	defer h.bp.Put(body)
-	err := json.NewEncoder(body).Encode(result)
-	if err != nil {
-		h.logger.Printf("E! failed to marshal row data json: %v", err)
-		return
-	}
+	// Technically we don't need the wait group, but
+	// having it hear will help contain any leaky go
+	// routines
+	var wg sync.WaitGroup
+	// We also didn't need these to be concurrent
+	for _, handler := range h.handlers {
+		wg.Add(1)
+		go func(handler *httppost.Handler) {
+			defer wg.Done()
+			body := h.bp.Get()
+			defer h.bp.Put(body)
+			err := json.NewEncoder(body).Encode(result)
+			if err != nil {
+				h.incrementErrorCount()
+				h.logger.Printf("E! failed to marshal row data json: %v", err)
+				return
+			}
+			req, err := handler.NewHTTPRequest(body)
+			if err != nil {
+				h.incrementErrorCount()
+				h.logger.Printf("E! failed to marshal row data json: %v", err)
+				return
+			}
 
-	resp, err := http.Post(h.url, "application/json", body)
-	if err != nil {
-		h.logger.Printf("E! failed to POST row data: %v", err)
-		return
+			req.Header.Set("Content-Type", "application/json")
+			resp, err := http.DefaultClient.Do(req)
```

Comment:
It hasn't been merged yet, but I also think that it won't break. The default client, I'm pretty sure, does proxy from environment.